logbook-netty/src/main/java/org/zalando/logbook/netty/Request.java

```diff
@@ -56,19 +64,29 @@ public String getMethod() {
 
     @Override
     public String getScheme() {
-        // TODO pick the real one
-        return "http";
+        SslHandler handler = context.channel().pipeline().get(SslHandler.class);
+        return handler == null ? "http" : "https";
     }
 
     @Override
     public String getHost() {
-        return request.headers().get(HOST, "unknown");
+        String host = request.headers().get(HOST);
+        if (host != null) {
+            int separator = host.indexOf(":");
+            return separator == -1 ? host : host.substring(0, separator);
+        } else {
+            return extractAddress().map(NetUtil::getHostname).orElse("unknown");
+        }
     }
 
     @Override
     public Optional<Integer> getPort() {
-        // TODO implement
-        return Optional.empty();
+        return extractAddress().map(InetSocketAddress::getPort);
+    }
+
+    private Optional<InetSocketAddress> extractAddress() {
+        SocketAddress address = origin == Origin.LOCAL ? context.channel().remoteAddress() : context.channel().localAddress();
```

Comment:
```suggestion
        final Channel channel = context.channel();
        final SocketAddress address = origin == LOCAL ? channel.remoteAddress() : channel.localAddress();
```

Reply:
Static import for Origin allows to use LOCAL directly.

Reply:
fixed